python/pyarrow/table.pxi

```diff
@@ -3549,6 +3549,29 @@ cdef class RecordBatch(_Tabular):
                                                                              row_major, pool))
         return pyarrow_wrap_tensor(c_tensor)
 
+    def copy_to(self, MemoryManager memory_manager):
```

Comment:
We could perhaps also accept a Device and copy to the Device's default memory manager, for usability.